src/array.cr

```diff
@@ -1430,6 +1430,62 @@ class Array(T)
     self
   end
 
+  # Returns `true` when array starts with *other*, otherwise `false`.
+  #
+  # ```
+  # a = [1, 2, 3]
+  # a.starts_with?([1, 2]) # => true
+  # a.starts_with?([2, 3]) # => false
+  # ```
+  def starts_with?(other : Array)
+    return false if other.size > size
+    self[0, other.size] == other
+  end
+
+  # ditto
+  def starts_with?(other : Enumerable)
+    starts_with?(other.to_a)
+  end
+  
+  # Returns `true` when array starts with *element*, otherwise `false`.
+  #
+  # ```
+  # a = [1, 2, 3]
+  # a.starts_with?(1) # => true
+  # a.starts_with?(2) # => false
+  # ```
+  def starts_with?(element : T)
+    return false if empty?
+    self[0] == element
+  end
+  
+  # Returns `true` when array ends with *other*, otherwise `false`.
+  # ```
+  # a = [1, 2, 3]
+  # a.ends_with?([2, 3]) # => true
+  # a.ends_with?([1, 2]) # => false
+  # ```
+  def ends_with?(other : Array)
+    return false if other.size > size
+    self[size - other.size, other.size] == other
+  end
+
+  # ditto
+  def ends_with?(other : Enumerable)
+    ends_with?(other.to_a)
+  end
+  
+  # Returns `true` when array ends with *element*, otherwise `false`.
+  # ```
+  # a = [1, 2, 3]
+  # a.ends_with?(3) # => true
+  # a.ends_with?(2) # => false
+  # ```
+  def ends_with?(element : T)
+    return false if empty?
+    self[-1] == element
```

Comment:
You can use the unsafe `unsafe_fetch` here since it won't be empty here. Ditto above.